components/builder-datastore/hooks/install

```diff
@@ -0,0 +1,496 @@
+    --timeout=$PG_TIMEOUT \
+    -w \
+    --silent
+
+  if postgres_healthy; then
+    echo PostgreSQL started successfully
+    return
+  else
+    echo PostgreSQL failed to started
+    return 1
+  fi
+}
+
+create_builder_database() {
+  echo Checking that the builder database exists in the PostgreSQL cluster
+  if builder_db_exists; then
+    echo The builder database already exists
+  else
+    echo Creating the builder database
+    sudo -u hab hab pkg exec "$ACTIVE_IDENT" createdb \
+      --host={{cfg.datastore.host}} \
+      --port={{cfg.datastore.port}} \
+      --username={{cfg.datastore.user}} \
+      --no-password \
+      --echo \
+      {{cfg.datastore.database}}
+  fi
+}
+
+stop_postgres() {
+
+  if ! postgres_running; then
+    echo PostgreSQL not running, will not issue stop
+    return
+  fi
+
+  echo "Stopping PostgreSQL (waiting ~${PG_TIMEOUT}s, current time: $(date +"%H:%M:%S"))"
+  sudo -u hab hab pkg exec "$ACTIVE_IDENT" pg_ctl -- stop \
+    --pgdata {{pkg.svc_data_path}} \
+    --mode=fast \
+    --timeout=$PG_TIMEOUT \
+    -w \
+    --silent
+
+  echo Allowing ~${PG_TIMEOUT}s for PostgreSQL to stop
+  for ((i = 0; i < PG_TIMEOUT; i++)); do
+    if postgres_running; then
+      echo "Waiting for PostgreSQL to stop ($((i + 1))/$PG_TIMEOUT)"
+       sleep 1
+    elif ((i >= PG_TIMEOUT)); then
+      echo "PostgreSQL failed to stop in ${PG_TIMEOUT}s"
+      echo "$output"
+      exit 13
+    else
+      echo PostgreSQL successfully stopped
+      return
+    fi
+  done
+}
+
+install_postgres() {
+  echo Installing PostgreSQL
+  create_postgres_cluster
+  start_postgres
+  create_builder_database
+  stop_postgres
+}
+
+manage_upgrade_readme() {
+  local -r readme="$1/README.txt"
+  if [[ ! -f "$readme" ]]; then
+    cat <<-EOF >"$readme"
+			This directory contains subdirectories used when upgrading PostgreSQL from
+			one version to another. The subdirectories are named using the date and time
+			of the upgrades. When you feel comfortable with the success of the upgrade
+			you may delete these files.
+		EOF
+  fi
+  echo "Created $readme, please review at your convenience"
+}
+
+postgres_needs_upgraded() {
+
+  echo Checking if PostgreSQL needs to be upgraded
+
+  # Prior to PostgreSQL 10 the version scheme was 3 parts (9.6.24, for example).
+  # Starting with PostgreSQL 10 the versioning scheme PostgreSQL versions are 2
+  # parts (for example, 17.4). We do have some 3 part 9.x.x PostgreSQL packages
+  # to account for migration from but given that 9.6.24 has been EOL since Nov
+  # 11th 2021 hopefully no one has a reason to continue using such packages in a
+  # context where they would be might migrate from from x.x.x to y.y.y.  Even if
+  # some did I think this will likely work acceptably
+  #
+  # https://www.postgresql.org/support/versioning/
+  # https://endoflife.date/postgresql
+
+  read -ra installed_version <<<"$(sudo -u hab hab pkg exec "$INSTALLED_PG_IDENT" pg_config -- --version | awk '{print $2}')"
```

Comment:
While upgrading (and this would probably be the case on a fresh install too), I get this error:
```
hab : user NOT in sudoers ; PWD=/hab/svc/builder-datastore ; USER=hab ; COMMAND=/usr/bin/hab pkg exec core/postgresql96 pg_config -- --version
```
This is going to fail on all sudo calls in the hook. This is run by the hab user that the supervisor uses for services which is not in sudoers so we need t ofigure out how to get this to work outside of sudo.

Reply:
All sudo usage has been removed and issues resulting have been dealt.  The most recent commit at the time this comment was added is dependent on https://github.com/habitat-sh/on-prem-builder/pull/284.  @sajjaphani and @agadgil-progress please review at your convenience tomorrow but I will be starting my day around 8 AM Eastern (5:30 PM IST) so I would love a round of feedback by then if possible.  CC: @rahulgoel1